utils/prometheus/prometheus.go

```diff
@@ -196,42 +187,29 @@ func (ph *Handler) GetSLIValue(metric string, start string, end string) (float64
 	if err != nil {
 		return 0, err
 	}
-	queryString := ph.ApiURL + "/api/v1/query?query=" + url.QueryEscape(query) + "&time=" + strconv.FormatInt(endUnix.Unix(), 10)
-	log.Println("GetSLIValue: Generated query: /api/v1/query?query=" + query + "&time=" + strconv.FormatInt(endUnix.Unix(), 10))
 
-	req, err := http.NewRequest("GET", queryString, nil)
-	req.Header.Set("Content-Type", "application/json")
+	log.Println("GetSLIValue: Generated query: /api/v1/query?query=" + query + "&time=" + strconv.FormatInt(endUnix.Unix(), 10))
 
-	resp, err := ph.HTTPClient.Do(req)
+	result, w, err := ph.API.Query(context.TODO(), query, endUnix)
+	if len(w) != 0 {
+		log.Printf("Prometheus API returned warnings: %v", w)
+	}
 	if err != nil {
 		return 0, err
 	}
-	defer resp.Body.Close()
 
-	body, _ := ioutil.ReadAll(resp.Body)
-	if resp.StatusCode != 200 {
-		return 0, errors.New("metric could not be received")
+	resultVector := result.(model.Vector)
```

Comment:
Please check the type assertion and handle the (albeit improbable) error :)

```suggestion
	resultVector, ok := result.(model.Vector)
	if !ok {
	    return 0, fmt.Errorf("prometheus response is not a Vector: %v", result)
	}
```